src/java/org/apache/cassandra/io/sstable/format/SSTableWriter.java

```diff
@@ -201,8 +201,8 @@ public void setOpenResult(boolean openResult)
     public SSTableReader finish(boolean openResult)
     {
         this.setOpenResult(openResult);
-        txnProxy.finish();
         observers.forEach(SSTableFlushObserver::complete);
+        txnProxy.finish();
```

Comment:
If we get a failure in `complete()`, and we've already called `finish()`, we'll get an illegal state...i.e. that we'll try to abort after committing due to the error from `complete()`.